backend/server/src/sync/harvest/client.rs

```diff
@@ -37,7 +39,12 @@ impl HarvestClient {
         } else {
             Scheme::HTTPS
         };
-        let authority = config.opencast.host.parse::<Authority>().expect("bug: invalid config");
+        let host = if config.opencast.host.parse::<Ipv6Addr>().is_ok() {
+            format!("[{}]", config.opencast.host)
+        } else {
+            config.opencast.host.clone()
+        };
+        let authority = host.parse::<Authority>().expect("bug: invalid config");
```

Comment:
This is now the third place that makes this distinction between `<IPv6Address>` and `[<IPv6Address>]`. I don't know how, but I feel like this should be handled in a more central fashion. :thinking:

Reply:
Mhhh what other places? Grepping for `Ipv6Addr` only shows this snippet.

Reply:
Somehow, reading my own comment again, it doesn't really make sense to me anymore. :sweat_smile: What I meant, though, I think, is that there are multiple places parsing this setting and applying some kind of logic to it. The other place being the `validate` function. I remember it taking me a relatively long time to comprehend these places fully. I don't know how to improve it in the code, either, though. I think it was also just that I was not fully aware of all the edge cases of the format, like sometimes IPv6 addresses need the `[]`.

Dunno, feel free to tell me I'm just too dumb, but I see an opportunity here to make things clearer; I just don't know how. xP Maybe you have an idea. If not, this shouldn't stop us from merging this PR, I guess.

Reply:
I think this might improve when we can switch to the config lib that allows us (in the future, anyway) to parse into sth different than `String`. So I would say ignore for now.